web/packages/core/src/ruffle-player.ts

```diff
@@ -165,9 +166,59 @@ export class RufflePlayer extends HTMLElement {
 
         this.ruffleConstructor = loadRuffle();
 
+        this.lastActivePlayingState = false;
+        this.setupPauseOnTabHidden();
+
         return this;
     }
 
+    /**
+     * Setup event listener to detect when tab is not active to pause instance playback.
+     * this.instance.play() is called when the tab becomes visible only if the 
+     * the instance was not paused before tab became hidden.
+     *  
+     * See: 
+     *      https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API
+     * @ignore
+     * @internal
+     */
+    setupPauseOnTabHidden(): void {
+        // Dynamically detect the property & event name for the current browser. 
+        let hidden: string | undefined;
+        let visibilityChange:string = "";
+        const documentAsAny: any = (document as any);
+        if (typeof documentAsAny.hidden !== "undefined") { // Opera 12.10 and Firefox 18 and later support
+          hidden = "hidden";
+          visibilityChange = "visibilitychange";
+        } else if (typeof documentAsAny.msHidden !== "undefined") {
+          hidden = "msHidden";
+          visibilityChange = "msvisibilitychange";
+        } else if (typeof documentAsAny.webkitHidden !== "undefined") {
+          hidden = "webkitHidden";
+          visibilityChange = "webkitvisibilitychange";
+        }
+
+        // Warn if the browser doesn't support addEventListener or the Page Visibility API
+        if (typeof document.addEventListener === "undefined" || hidden === undefined) {
+          console.log("The Page Visibility API is not supported in this browser. Audio may be out of sync when switching tabs.");
+        } else {
```

Comment:
These support checks are overkill because if we're on a browser that doesn't support document.addEventListener, we'd fail much earlier than this point. That is, anything that doesn't support [Page Visibility](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event#browser_compatibility) already doesn't support [WebAssembly](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/instantiate) or Shadow DOM or etc., so it seems fine to just expect the standard API, and not check for prefixes.